### ISSUE TITLE
Unable to start gitea service any which way I try

### ISSUE THREAD

**Issue by webhype** [go-gitea/gitea#17869]
### Gitea Version

Gitea version 1.15.6 built with GNU Make 4.1, go1.16.9 : bindata, sqlite, sqlite_unlock_notify (installed 30 Nov 2021)

### Git Version

2.33.0

### Operating System

Ubuntu 16.04.7 LTS (GNU/Linux 4.4.0-210-generic x86_64)

### How are you running Gitea?

gitea (latest binary download) starts fine manually when I am root and use `GITEA_WORK_DIR=/var/lib/gitea/ /usr/local/bin/gitea web -c /etc/gitea/app.ini` . I was able to enter the app via localhost:3000, connect to the database, create a user, and so on.

Any attempts to get to start it as a service fail like this:
```
$ systemctl restart gitea
$ systemctl status gitea
* gitea.service - Gitea (Git with a cup of tea)
   Loaded: loaded (/etc/systemd/system/gitea.service; enabled; vendor preset: enabled)
   Active: activating (auto-restart) (Result: exit-code) since Tue 2021-11-30 23:50:09 UTC; 1s ago
  Process: 14226 ExecStart=/usr/local/bin/gitea web --config /etc/gitea/app.ini (code=exited, status=1/FAILURE)
 Main PID: 14226 (code=exited, status=1/FAILURE)
Nov 30 23:50:09 HUB1 systemd[1]: gitea.service: Main process exited, code=exited, status=1/FAILURE
Nov 30 23:50:09 HUB1 systemd[1]: gitea.service: Unit entered failed state.
Nov 30 23:50:09 HUB1 systemd[1]: gitea.service: Failed with result 'exit-code'.
```

### Database

MySQL

### Can you reproduce the bug on the Gitea demo site?

No

### Description

NB: User `git` was already taken so I use user `gitea` (`/home/gitea` and so on).

### What I have tried that didn't work

`chown -R gitea:gitea /var/lib/gitea`
`chown gitea:gitea /usr/local/bin/gitea`

None of it has any effect. There is absolutely no information in the logs as to why it would fail w/ error 1.

`/etc/gitea/app.ini` file:

```ini
APP_NAME = Gitea: Git with a cup of tea
RUN_USER = gitea
RUN_MODE = prod

[security]
INTERNAL_TOKEN     = REDACTED
INSTALL_LOCK       = true
SECRET_KEY         = REDACTED
PASSWORD_HASH_ALGO = pbkdf2

[database]
DB_TYPE  = mysql
HOST     = 127.0.0.1:3306
NAME     = giteadb
USER     = gitea
PASSWD   = REDACTED
SCHEMA   = 
SSL_MODE = disable
CHARSET  = utf8
PATH     = /var/lib/gitea/data/gitea.db
LOG_SQL  = false

[repository]
ROOT = /var/lib/gitea/data/gitea-repositories

[server]
SSH_DOMAIN       = localhost
DOMAIN           = localhost
HTTP_PORT        = 3000
ROOT_URL         = http://localhost:3000/
DISABLE_SSH      = false
SSH_PORT         = 22
LFS_START_SERVER = true
LFS_CONTENT_PATH = /var/lib/gitea/data/lfs
LFS_JWT_SECRET   = READACTED
OFFLINE_MODE     = false

[mailer]
ENABLED = false

[service]
REGISTER_EMAIL_CONFIRM            = false
ENABLE_NOTIFY_MAIL                = false
DISABLE_REGISTRATION              = false
ALLOW_ONLY_EXTERNAL_REGISTRATION  = false
ENABLE_CAPTCHA                    = false
REQUIRE_SIGNIN_VIEW               = false
DEFAULT_KEEP_EMAIL_PRIVATE        = false
DEFAULT_ALLOW_CREATE_ORGANIZATION = true
DEFAULT_ENABLE_TIMETRACKING       = true
NO_REPLY_ADDRESS                  = noreply.localhost

[picture]
DISABLE_GRAVATAR        = false
ENABLE_FEDERATED_AVATAR = true

[openid]
ENABLE_OPENID_SIGNIN = true
ENABLE_OPENID_SIGNUP = true

[session]
PROVIDER = file

[log]
MODE      = error_console
LEVEL     = Trace
ROOT_PATH = /var/lib/gitea/log
ROUTER    = console

[Log.error_console]
Mode=console
Stderr=true
Level=Trace
```

`/etc/systemd/system/gitea.service` file:

```systemd
[Unit]
Description=Gitea (Git with a cup of tea)
After=syslog.target
After=network.target
###
# Don't forget to add the database service dependencies
###
#
Wants=mysql.service
After=mysql.service
#
#Wants=mariadb.service
#After=mariadb.service
#
#Wants=postgresql.service
#After=postgresql.service
#
#Wants=memcached.service
#After=memcached.service
#
#Wants=redis.service
#After=redis.service
#
###
# If using socket activation for main http/s
###
#
#After=gitea.main.socket
#Requires=gitea.main.socket
#
###
# (You can also provide gitea an http fallback and/or ssh socket too)
#
# An example of /etc/systemd/system/gitea.main.socket
###
##
## [Unit]
## Description=Gitea Web Socket
## PartOf=gitea.service
##
## [Socket]
## Service=gitea.service
## ListenStream=<some_port>
## NoDelay=true
##
## [Install]
## WantedBy=sockets.target
##
###

[Service]
# Modify these two values and uncomment them if you have
# repos with lots of files and get an HTTP error 500 because
# of that
###
#LimitMEMLOCK=infinity
#LimitNOFILE=65535
RestartSec=2s
Type=simple
User=gitea
Group=gitea
WorkingDirectory=/var/lib/gitea/
# If using Unix socket: tells systemd to create the /run/gitea folder, which will contain the gitea.sock file
# (manually creating /run/gitea doesn't work, because it would not persist across reboots)
#RuntimeDirectory=gitea
ExecStart=/usr/local/bin/gitea web --config /etc/gitea/app.ini
Restart=always
Environment=USER=gitea HOME=/home/gitea GITEA_WORK_DIR=/var/lib/gitea
# If you install Git to directory prefix other than default PATH (which happens
# for example if you install other versions of Git side-to-side with
# distribution version), uncomment below line and add that prefix to PATH
# Don't forget to place git-lfs binary on the PATH below if you want to enable
# Git LFS support
#Environment=PATH=/usr/bin/git:/bin:/sbin:/usr/bin:/usr/sbin
# If you want to bind Gitea to a port below 1024, uncomment
# the two values below, or use socket activation to pass Gitea its ports as above
###
#CapabilityBoundingSet=CAP_NET_BIND_SERVICE
#AmbientCapabilities=CAP_NET_BIND_SERVICE
###

[Install]
WantedBy=multi-user.target
```

### Typical Logs

```
Dec 01 00:14:33 HUB1 systemd[1]: Started Gitea (Git with a cup of tea).
Dec 01 00:14:33 HUB1 gitea[13794]: 2021/12/01 00:14:33 cmd/web.go:102:runWeb() [I] Starting Gitea on PID: 13794
Dec 01 00:14:33 HUB1 gitea[13794]: 2021/12/01 00:14:33 cmd/web.go:146:runWeb() [I] Global init
Dec 01 00:14:33 HUB1 gitea[13794]: 2021/12/01 00:14:33 routers/init.go:74:GlobalInit() [I] Git Version: 2.33.0, Wire Protocol Version 2 Enabled
Dec 01 00:14:33 HUB1 gitea[13794]: 2021/12/01 00:14:33 routers/init.go:77:GlobalInit() [I] AppPath: /usr/local/bin/gitea
Dec 01 00:14:33 HUB1 gitea[13794]: 2021/12/01 00:14:33 routers/init.go:78:GlobalInit() [I] AppWorkPath: /var/lib/gitea
Dec 01 00:14:33 HUB1 gitea[13794]: 2021/12/01 00:14:33 routers/init.go:79:GlobalInit() [I] Custom path: /var/lib/gitea/custom
Dec 01 00:14:33 HUB1 gitea[13794]: 2021/12/01 00:14:33 routers/init.go:80:GlobalInit() [I] Log path: /var/lib/gitea/log
Dec 01 00:14:33 HUB1 gitea[13794]: 2021/12/01 00:14:33 routers/init.go:81:GlobalInit() [I] Run Mode: Prod
Dec 01 00:14:35 HUB1 systemd[1]: gitea.service: Unit entered failed state.
Dec 01 00:14:35 HUB1 systemd[1]: gitea.service: Failed with result 'exit-code'.
```

This is only the logs I could gather from `systemctl restart gitea` etc. `/var/lib/gitea/log` is an empty directory.

**Comment by zeripath** [comment]
Something is killing gitea extremely early on.

When you say "latest" version - EXACTLY what version do you mean? 

Is there any information in journald? Is there any other logging? 

It would sensible to simplify your log configuration to what is recommended in our documentation:

```ini
[log]
MODE = console
LEVEL = debug ; please set the level to debug when we are debugging a problem
ROUTER = console
COLORIZE = false ; this can be true if you can strip out the ansi coloring
```

If you are starting with systemd you will find then that the logs will go to journald or syslog.

**Comment by wxiaoguang** [comment]
Another possible method: try to run Gitea in command line directly, with the same config, without systemd, to see what happens.

**Comment by lunny** [comment]
> Another possible method: try to run Gitea in command line directly, with the same config, without systemd, to see what happens.

`su gitea` and try run it in console. And change the directory owner to gitea:gitea `chown -R gitea:gitea /var/lib/gitea`

**Comment by webhype** [comment]
Thanks for the answers, guys.

TL;DR: ` chown -R gitea:gitea /var/lib/gitea` for a 2nd time seems to have fixed it.

`su gitea` was the breakthrough tip!

```
gitea@HUB1:~$ GITEA_WORK_DIR=/var/lib/gitea/ /usr/local/bin/gitea web -c /etc/gitea/app.ini
2021/12/02 22:38:38 cmd/web.go:102:runWeb() [I] Starting Gitea on PID: 32324
2021/12/02 22:38:38 cmd/web.go:146:runWeb() [I] Global init
2021/12/02 22:38:38 routers/init.go:74:GlobalInit() [I] Git Version: 2.33.0, Wire Protocol Version 2 Enabled
2021/12/02 22:38:38 routers/init.go:77:GlobalInit() [I] AppPath: /usr/local/bin/gitea
2021/12/02 22:38:38 routers/init.go:78:GlobalInit() [I] AppWorkPath: /var/lib/gitea/
2021/12/02 22:38:38 routers/init.go:79:GlobalInit() [I] Custom path: /var/lib/gitea/custom
2021/12/02 22:38:38 routers/init.go:80:GlobalInit() [I] Log path: /var/lib/gitea/log
2021/12/02 22:38:38 routers/init.go:81:GlobalInit() [I] Run Mode: Prod
2021/12/02 22:38:38 ...dules/setting/log.go:286:newLogService() [I] Gitea v1.15.6 built with GNU Make 4.1, go1.16.9 : bindata, sqlite, sqlite_unlock_notify
2021/12/02 22:38:38 ...dules/setting/log.go:333:newLogService() [I] Gitea Log Mode: Console(Console:)
2021/12/02 22:38:38 ...dules/setting/log.go:249:generateNamedLogger() [I] Router Log: Console(console:)
2021/12/02 22:38:38 ...les/setting/cache.go:78:newCacheService() [I] Cache Service Enabled
2021/12/02 22:38:38 ...les/setting/cache.go:93:newCacheService() [I] Last Commit Cache Service Enabled
2021/12/02 22:38:38 ...s/setting/session.go:77:newSessionService() [I] Session Service Enabled
2021/12/02 22:38:38 ...s/storage/storage.go:171:initAttachments() [I] Initialising Attachment storage with type: 
2021/12/02 22:38:38 ...les/storage/local.go:47:NewLocalStorage() [I] Creating new Local Storage at /var/lib/gitea/data/attachments
2021/12/02 22:38:38 ...s/storage/storage.go:165:initAvatars() [I] Initialising Avatar storage with type: 
2021/12/02 22:38:38 ...les/storage/local.go:47:NewLocalStorage() [I] Creating new Local Storage at /var/lib/gitea/data/avatars
2021/12/02 22:38:38 ...s/storage/storage.go:183:initRepoAvatars() [I] Initialising Repository Avatar storage with type: 
2021/12/02 22:38:38 ...les/storage/local.go:47:NewLocalStorage() [I] Creating new Local Storage at /var/lib/gitea/data/repo-avatars
2021/12/02 22:38:38 ...s/storage/storage.go:177:initLFS() [I] Initialising LFS storage with type: 
2021/12/02 22:38:38 ...les/storage/local.go:47:NewLocalStorage() [I] Creating new Local Storage at /var/lib/gitea/data/lfs
2021/12/02 22:38:38 ...s/storage/storage.go:189:initRepoArchives() [I] Initialising Repository Archive storage with type: 
2021/12/02 22:38:38 ...les/storage/local.go:47:NewLocalStorage() [I] Creating new Local Storage at /var/lib/gitea/data/repo-archive
2021/12/02 22:38:38 ...ueue_disk_channel.go:137:Run() [D] PersistableChannelQueue: push_update Starting
2021/12/02 22:38:38 ...eue/queue_wrapped.go:73:setInternal() [W] [Attempt: 1] Failed to create queue: level for push_update cfg: queue.LevelQueueConfiguration{ByteFIFOQueueConfiguration:queue.ByteFIFOQueueConfiguration{WorkerPoolConfiguration:queue.WorkerPoolConfiguration{QueueLength:20, BatchLength:20, BlockTimeout:1000000000, BoostTimeout:300000000000, BoostWorkers:1, MaxWorkers:5}, Workers:0, Name:"push_update-level", WaitOnEmpty:false}, DataDir:"/var/lib/gitea/data/queues/common", ConnectionString:"", QueueName:""} error: open /var/lib/gitea/data/queues/common/LOCK: permission denied
2021/12/02 22:38:38 ...ueue_disk_channel.go:137:Run() [D] PersistableChannelQueue: notification-service Starting
2021/12/02 22:38:38 ...eue/queue_wrapped.go:73:setInternal() [W] [Attempt: 1] Failed to create queue: level for notification-service cfg: queue.LevelQueueConfiguration{ByteFIFOQueueConfiguration:queue.ByteFIFOQueueConfiguration{WorkerPoolConfiguration:queue.WorkerPoolConfiguration{QueueLength:20, BatchLength:20, BlockTimeout:1000000000, BoostTimeout:300000000000, BoostWorkers:1, MaxWorkers:5}, Workers:0, Name:"notification-service-level", WaitOnEmpty:false}, DataDir:"/var/lib/gitea/data/queues/common", ConnectionString:"", QueueName:""} error: open /var/lib/gitea/data/queues/common/LOCK: permission denied
2021/12/02 22:38:38 ...ueue_disk_channel.go:162:Run() [D] PersistableChannelUniqueQueue: repo-archive Starting
2021/12/02 22:38:38 ...eue/queue_wrapped.go:73:setInternal() [W] [Attempt: 1] Failed to create queue: unique-level for repo-archive cfg: queue.LevelUniqueQueueConfiguration{ByteFIFOQueueConfiguration:queue.ByteFIFOQueueConfiguration{WorkerPoolConfiguration:queue.WorkerPoolConfiguration{QueueLength:20, BatchLength:20, BlockTimeout:1000000000, BoostTimeout:300000000000, BoostWorkers:1, MaxWorkers:5}, Workers:0, Name:"repo-archive-level", WaitOnEmpty:false}, DataDir:"/var/lib/gitea/data/queues/common", ConnectionString:"", QueueName:""} error: open /var/lib/gitea/data/queues/common/LOCK: permission denied
2021/12/02 22:38:38 routers/init.go:93:GlobalInit() [I] SQLite3 Supported
2021/12/02 22:38:38 routers/common/db.go:20:InitDBEngine() [I] Beginning ORM engine initialization.
2021/12/02 22:38:38 routers/common/db.go:27:InitDBEngine() [I] ORM engine initialization attempt #1/10...
2021/12/02 22:38:38 ...om/urfave/cli/app.go:524:HandleAction() [I] PING DATABASE mysql
2021/12/02 22:38:39 ...om/urfave/cli/app.go:524:HandleAction() [W] Table project column board_type db type is INT(10) UNSIGNED, struct type is INT UNSIGNED
2021/12/02 22:38:39 ...om/urfave/cli/app.go:524:HandleAction() [W] Table project column type db type is INT(10) UNSIGNED, struct type is INT UNSIGNED
2021/12/02 22:38:39 routers/init.go:98:GlobalInit() [I] ORM engine initialization successful!
2021/12/02 22:38:39 routers/init.go:104:GlobalInit() [F] Failed to initialize OAuth2 support: Error while loading or creating symmetric key: open /var/lib/gitea/data/jwt/private.pem: permission denied
```

These `permission denied` errors prompted me to AGAIN issue ` chown -R gitea:gitea /var/lib/gitea` -- even though I did so several times before yesterday and it seemed to have no effect?!?

Then it worked in manual mode (`GITEA_WORK_DIR=/var/lib/gitea/ /usr/local/bin/gitea web -c /etc/gitea/app.ini`).

Then I restarted the service and "it just worked".

One odd thing though: 

Starting gitea manually *and then pressing `Ctrl+C` leaves port 3000 listening for http request, and serves a homepage.* I.e. it doesn't work like starting, say, Apache httpd manually, where `Ctrl+C` terminates all services.

Even after exiting out of a manual start, I have to always manually kill whatever is listening port 3000 (using ` lsof -n -i :3000 | grep LISTEN | awk "// {print \$2;}"` to get the pid, then `kill PID`).

Is this expected behavior or an oddity of my install? Does a manually-started gitea somehow spin-off the webserver as a separate thread?

**Comment by webhype** [comment]
Ok I basically did _everything_ suggested here, changed logging to debug,  re-applied `chown -R gitea:gitea /var/lib/gitea` for what must have been the 10th time, executed `su gitea`, then manually started it via `GITEA_WORK_DIR=/var/lib/gitea/ /usr/local/bin/gitea web -c /etc/gitea/app.ini` (which worked) and then restarted the service, and it worked for some reason.

What was wrong in my original configuration? Who knows! On the upside, anyone confused can use the above config as a model that works lol. Thanks again for all responses!
